core/shared/src/main/scala/scalaz/zio/Task.scala

```diff
@@ -0,0 +1,631 @@
+
+  type TaskE = Throwable
+  type TaskR = Any
+
+  /*
+   * ----------------------------------------------------------------------------------------------------------------------
+   * The Following is taken from ZIOFunctions.
+   * ----------------------------------------------------------------------------------------------------------------------
+   */
+  //  // ALL error types in this trait must be a subtype of `UpperE`.
+  //  type UpperE = Throwable
+  //  // ALL environment types in this trait must be a supertype of `LowerR`.
+  //  type LowerR = Any
+
+  /** Returns an effect that models failure with the specified error.
+   * The moral equivalent of `throw` for pure code.
+   * [[scalaz.zio.ZIO.fail]]
+   */
+  final def fail(error: TaskE): IO[TaskE, Nothing] = ZIO.fail(error)
+
+  /** Returns an effect that models failure with the specified `Cause`.
+   *
+   *  [[scalaz.zio.ZIO.halt]]
+   */
+  final def halt(cause: Cause[TaskE]): IO[TaskE, Nothing] = ZIO.halt(cause)
+
+  /** Returns an effect that models success with the specified strictly-
+   * evaluated value.
+   *
+   * [[scalaz.zio.ZIO.succeed]]
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** Returns an effect that models success with the specified lazily-evaluated
+   * value. This method should not be used to capture effects. See
+   * `[[ZIO.effectTotal]]` for capturing total effects, and `[[ZIO.effect]]` for capturing
+   * partial effects.
+   *
+   * [[scalaz.zio.ZIO.succeedLazy]]
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** Accesses the whole environment of the effect.
+   *
+   *  [[scalaz.zio.ZIO.environment]]
+   */
+  final def environment: ZIO[TaskR, Nothing, Any] = ZIO.access(ZIO.identityFn[TaskR])
+
+  /** Accesses the environment of the effect.
+   * {{{
+   * val portNumber = effect.access(_.config.portNumber)
+   * }}}
+   *
+   * [[scalaz.zio.ZIO.access]]
+   */
+  final def access: ZIO.AccessPartiallyApplied[Any] =
+    ZIO.access
+
+  /** Effectfully accesses the environment of the effect.
+   *
+   *  [[scalaz.zio.ZIO.accessM]]
+   */
+  final def accessM: ZIO.AccessMPartiallyApplied[Any] =
+    ZIO.accessM
+
+  /** Given an environment `R`, returns a function that can supply the
+   * environment to programs that require it, removing their need for any
+   * specific environment.
+   *
+   * This is similar to dependency injection, and the `provide` function can be
+   * thought of as `inject`.
+   *
+   * [[scalaz.zio.ZIO.provide]]
+   */
+  final def provide[A](r: Any): ZIO[TaskR, TaskE, A] => IO[TaskE, A] =
+    ZIO.provide(r)
+
+  /** Returns an effect that accesses the runtime, which can be used to
+   * (unsafely) execute tasks. This is useful for integration with
+   * non-functional code that must call back into functional code.
+   *
+   * [[scalaz.zio.ZIO.runtime]]
+   */
+  final def runtime: ZIO[TaskR, Nothing, Runtime[TaskR]] = ZIO.runtime
```

Comment:
I'd replace all the `ZIO[TaskR, ...]` by either `Task` (if the error is `Throwable`) or `UIO` (if the error is `Nothing`).